uniffi/src/templates/ObjectTemplate.rs

```diff
@@ -42,21 +35,13 @@ lazy_static::lazy_static! {
 {%- for meth in obj.methods() %}
     #[no_mangle]
     pub extern "C" fn {{ meth.ffi_func().name() }}(
-        {%- for arg in meth.ffi_func().arguments() %}
-        {{ arg.name() }}: {{ arg.type_()|type_c }},
-        {%- endfor %}
-    ) -> {% match meth.ffi_func().return_type() %}{% when Some with (return_type) %}{{ return_type|ret_type_c }}{% else %}(){% endmatch %} {
+        {%- call rs::arg_list_rs_decl(meth.ffi_func().arguments()) %}) -> {% match meth.ffi_func().return_type() %}{% when Some with (return_type) %}{{ return_type|ret_type_c }}{% else %}(){% endmatch %} {
```

Comment:
I wonder if the return type can also be a macro, although you don't need to add it here, I can add it in my effort in #31

Also, this might be a dumb question, but is there a reason why we do the matching here in the object template, but we do a match before the functions on the TopLevelFunction template? Not that I have an opinion on which is really better, but just wondering

Reply:
I don't like the way we have to duplicate the function body, but have sacrificed de-duplication in favour of clarity/readablity.

Then I saw what you were doing in #31 with the `return_type` macro, and thought it a better way.

I wanted to land this work to make it a base for you to work on and improve in #31.